File: javascript/space-age/space-age.js
class SpaceAge {
  constructor(seconds) {
    this.seconds = seconds;
  }
  years(period = 1) {
    return Math.round((this.seconds / (31557600 * period)) * 100) / 100;
  }
  onEarth() {
    return this.years();
  }
  onMercury() {
    return this.years(0.2408467);
  }
  onVenus() {
    return this.years(0.61519726);
  }
  onMars() {
    return this.years(1.8808158);
  }
  onJupiter() {
    return this.years(11.862615);
  }
  onSaturn() {
    return this.years(29.447498);
  }
  onUranus() {
    return this.years(84.016846);
  }
  onNeptune() {
    return this.years(164.79132);
  }
}
module.exports = SpaceAge;

File: ruby/space-age/space_age.rb
class SpaceAge 
  def initialize(secs)
    @secs = secs
  end

  def years(period = 1) 
    ((@secs * 1000 / (31557600 * period) * 1000) ).to_f / 1000000
  end 

  def on_earth
    years
  end

  def on_mercury
    years(0.2408467)
  end

  def on_venus
    years(0.61519726)
  end

  def on_mars
    years(1.8808158)
  end

  def on_jupiter
    years(11.862615)
  end

  def on_saturn
    years(29.447498)
  end

  def on_uranus
    years(84.016846)
  end

  def on_neptune
    years(164.79132)
  end
end

module BookKeeping
  VERSION = 1
end